olive/engine/packaging/packaging_generator.py

```diff
@@ -89,9 +89,13 @@ def _package_candidate_models(
                     model_config = pf_footprint.get_model_config(model_id)
                     onnx_file_name = model_config.get("onnx_file_name")
                     onnx_model = ONNXModel(temp_resource_path, onnx_file_name)
+                    model_name = Path(onnx_model.model_path).name
                     for file in Path(temp_resource_path.get_path()).iterdir():
-                        if file.name == Path(onnx_model.model_path).name:
-                            file_name = "model.onnx"
+                        if file.name.startswith(model_name):
+                            if file.name.endswith(".onnx"):
+                                file_name = "model.onnx"
+                            elif file.name.endswith(".onnx.data"):
```

Comment:
Do you have an example of the failure? 
I think the previous logic should have worked since we are only renaming the `.onnx` file to `model.onnx` but keeping the original name for the other files. 
onnx external data doesn't work if the file name is changed. This logic assumes that the external data is `model.onnx.data`. Which is true most of the time but is not always true if user provided their own name during conversion.

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/39bb21ce-695b-4459-87d6-aa518818a06e)
I see, looks like *.onnx.data will be recorded in model's proto info.

The bug is caused by line 93 where we will rename(move) the original model. When it comes two 2rd iteration on `x.onnx.data`, the `onnx_model.model_path` will fail.
![image](https://github.com/microsoft/Olive/assets/13343117/a184aa47-38cc-4fec-ad0e-ff71a758a6df)
![image](https://github.com/microsoft/Olive/assets/13343117/d6aa0cb8-0666-4f34-b6dd-33b7c5957d26)

Reply:
I just updated with temp model path, please help take a look.